### ISSUE TITLE
Mod Manager crash when clicking on Save And Play

### ISSUE THREAD

**Issue by Tartellu** [X-Hax/SA-Mod-Manager#108]
## Exception Report
```
SAMM Information:
	Manager Version: 1.2.6
	Current Game: Sonic Adventure 2
	Game Directory: C:\Program Files (x86)\Steam\steamapps\common\Sonic Adventure 2
	Manager Location: C:\Users\joe\Desktop\
	Running as Admin: False
	Mod Loader Version (Hash): e23b7f10705afedf4e562625f2f05b69ac23a82b
```

```
Exception Information:
	Type: Win32Exception
	Exception: An error occurred trying to start process 'Launcher.exe' with working directory 'C:\Windows\system32'. The system cannot find the file specified.
	Source: System.Diagnostics.Process
	In Function: Boolean StartWithShellExecuteEx(System.Diagnostics.ProcessStartInfo)
	StackTrace:    at System.Diagnostics.Process.StartWithShellExecuteEx(ProcessStartInfo startInfo)
	   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
	   at SAModManager.MainWindow.StartGame()
	   at SAModManager.MainWindow.SaveAndPlayButton_Click(Object sender, RoutedEventArgs e)
	   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
	   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
	   at System.Windows.Controls.Primitives.ButtonBase.OnClick()
	   at System.Windows.Controls.Button.OnClick()
	   at System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
	   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
	   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
	   at System.Windows.UIElement.ReRaiseEventAs(DependencyObject sender, RoutedEventArgs args, RoutedEvent newEvent)
	   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
	   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
	   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
	   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
	   at System.Windows.Input.InputManager.ProcessStagingArea()
	   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
	   at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
	   at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
	   at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
	   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
	   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
	   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
	   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
	   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
	   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
	   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
	   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
	   at System.Windows.Application.RunDispatcher(Object ignore)
	   at System.Windows.Application.RunInternal(Window window)
	   at SAModManager.App.Main()
```

**Comment by Sora-yx** [comment]
I don't know what you did, but there is nothing right with this set up, your game location is in **program files** but somehow it looks for "launcher.exe" in system32? 
Anyway, best move is to reset the Manager, go in AppData -> Local and delete the folder SAManager.

**Comment by Sora-yx** [comment]
Do you have sonic2app.exe? The error mentions launcher.exe, but the  Manager originally uses sonic2app to start the game, if you renamed it, it won't work.

**Comment by Tartellu** [comment]
yes, I validated my files and replaced it. The old mod manager would strangely boot into the sonic2app but would take forever to load once I got into the game, the new one won't even do the for me.

**Comment by Sora-yx** [comment]
Can you try to manually re set the game location in the Manager Config tab?

**Comment by Tartellu** [comment]
Yes, I moved the game files to my desktop and it still crashes.

**Comment by ItsEasyActually** [comment]
Found the issue, it was related to custom executables included with mods. Whatever mod you're using that did this is probably not something you should be using. Regardless, thanks to this report, we found the bug and got it fixed as of [bf7be21](https://github.com/X-Hax/SA-Mod-Manager/commit/bf7be21475f96c48969bb57be974f00ead86c60e).
